### PR TITLE
Enable strategies written directly in Python as an alt to engines. (duplicate)

### PR DESCRIPTION
**Pull request by icecream17**
Same as #187 but updated. 

Notable difference between this pr and #187 is that instead of a strategy function, you implement the whole class.
Which enables more functionality but takes more setup.

Adds docs folder (only 1 file)

Breaking change: Previously, if the protocol was not "xboard" it was assumed to be "uci".
But now if someone put something like "yci" it throws

### CODE REVIEW

File: docs/creating-a-custom-bot.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Creating a custom bot
```
Comment (TheYoBots) [inline docs/creating-a-custom-bot.md:1]: Would be nice if this file is in the main README.md file. But make the change at own will.

**Review by TheYoBots — CHANGES_REQUESTED** [review]

**Comment by TheYoBots** [comment]
Also maybe linking something that shows a bunch of strategies written in python would allow further usage of this ‘protocol’ and it won’t be less extensively used [referring to https://github.com/ShailChoksi/lichess-bot/pull/187#issuecomment-792349236]

File: engine_wrapper.py [hunk 2]
```diff
@@ -148,3 +156,8 @@ def get_opponent_info(self, game):
             self.engine.protocol.send_line(f"rating {game.me.rating} {game.opponent.rating}")
         if game.opponent.title == "BOT":
             self.engine.protocol.send_line("computer")
+
+def getHomemadeEngine():
+    raise NotImplementedError(
+        "    You haven't changed the getHomemadeEngine function yet!\n"
+        "    See docs/creating-a-custom-bot.md")
```
Comment (TheYoBots) [inline engine_wrapper.py:163]: I think after the request I made, you had forgotten to change this line.

**Review by TheYoBots — CHANGES_REQUESTED** [review]

**Comment by TheYoBots** [comment]
Does an engine have to be added for this to function? It would be nice if the engine was disabled all together, though I’m not sure of any way to do this.

**Comment by icecream17** [comment]
If there's not a bug, a class only has to implement `search` and extend `MinimalEngine`:

```py
# strategies.py
class ExampleEngine(MinimalEngine):
    pass

class RandomMove(ExampleEngine):
    def search(self, board, *args):
        return random.choice(list(board.legal_moves))
```

**Comment by MarkZH** [comment]
> If there's not a bug, a class only has to implement `search` and extend `MinimalEngine`:
> 
> ```python
> # strategies.py
> class ExampleEngine(MinimalEngine):
>     pass
> 
> class RandomMove(ExampleEngine):
>     def search(self, board, *args):
>         return random.choice(list(board.legal_moves))
> ```
> 
> Although `MinimalEngine` itself sets `self.engine = self` because of some methods in `EngineWrapper`

`name()` needs to be overridden, too.

**Review by ShailChoksi — APPROVED** [review]

**Pull request merged**